### PR TITLE
refactor: remove the model from fragment

### LINKED ISSUE #1560: Refactor the code in Settings Fragment
**Actual Behaviour**

Currently the settings fragment uses model inside the view fragment only and not the presenter.

**Expected Behaviour**

The code should be refactored according to the MVP architecture here.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1560 

Changes: made two functions to set the login and logout preferences instead of checking them using if and else in the fragment.

### CODE REVIEW

**Comment by codedsun** [comment]
@batbrain7 Please update the branch

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt [hunk 1]
```diff
@@ -388,4 +301,76 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
...
+
+        /*   set Mic preferences    */
+        micSettings.setOnPreferenceClickListener {
+            settingsPresenter.sendSetting(Constant.MIC_INPUT, (PrefManager.getBoolean(Constant.MIC_INPUT, false)).toString(), 1)
+            true
+        }
+
+        /* logout on clicking loginLogout */
+        loginLogout.setOnPreferenceClickListener {
+            val d = AlertDialog.Builder(activity as SkillsActivity)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: Single letter variable names

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt [hunk 2]
```diff
@@ -30,13 +29,6 @@ import org.fossasia.susi.ai.skills.settings.contract.ISettingsPresenter
 import org.fossasia.susi.ai.skills.settings.contract.ISettingsView
 import timber.log.Timber
 
-
-/**
- * The Fragment for Settings Activity
- *
- * Created by mayanktripathi on 10/07/17.
- */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/SettingsPresenter.kt [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
 import org.fossasia.susi.ai.skills.settings.contract.ISettingsPresenter
 import org.fossasia.susi.ai.skills.settings.contract.ISettingsView
 import retrofit2.Response
 
-/**
- * Presenter for Settings
- * The P in MVP
- *
- * Created by mayanktripathi on 07/07/17.
- */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/SettingsPresenter.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/contract/ISettingsPresenter.kt [hunk 4]
```diff
@@ -1,11 +1,5 @@
 package org.fossasia.susi.ai.skills.settings.contract
 
-/**
- * The interface for Settings Presenter
- *
- * Created by mayanktripathi on 07/07/17.
- */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/contract/ISettingsPresenter.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/contract/ISettingsView.kt [hunk 5]
```diff
@@ -1,11 +1,5 @@
 package org.fossasia.susi.ai.skills.settings.contract
 
-/**
- * The interface for Settings view
- *
- * Created by mayanktripathi on 07/07/17.
- */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/contract/ISettingsView.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/SettingsPresenter.kt [hunk 6]
```diff
@@ -145,5 +140,12 @@ class SettingsPresenter(skillsActivity: SkillsActivity) : ISettingsPresenter, IS
         settingView?.setServerSuccessful()
     }
 
+    override fun checkLogin() {
+        if (PrefManager.getToken() == null) {
+            settingView?.setLogoutPreferences(utilModel.getString(R.string.not_logged_in), utilModel.getString(R.string.action_login))
+        } else {
+            settingView?.setLoginPreferences(PrefManager.getStringSet(Constant.SAVED_EMAIL).iterator().next().toString(), utilModel.getString(R.string.action_log_out))
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/SettingsPresenter.kt]: Please simplify this a bit. We need not use the string set and iterator here.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt [hunk 7]
```diff
@@ -390,4 +303,76 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
         super.onDestroyView()
         settingsPresenter.onDetach()
     }
+
+    override fun setLoginPreferences(message: String, loginMessage: String) {
+        loginLogout.title = loginMessage
+        displayEmail.title = message
+
+        /*   set Mic preferences    */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: A single line comment.
```suggestion
        // Set mic preferences
```

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt (hunk 2 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: well this should not be present in a file, so it is best to remove these while editing each file
